src/client/Client.cc

```diff
@@ -5440,8 +5440,16 @@ int Client::may_setattr(Inode *in, struct ceph_statx *stx, int mask,
   }
 
   if (mask & CEPH_SETATTR_MODE) {
-    if (perms.uid() != 0 && perms.uid() != in->uid)
-      goto out;
+    if (perms.uid() != 0 && perms.uid() != in->uid) {
+      if((in->mode & S_ISUID) || (in->mode & S_ISGID)) {
```

Comment:
this means that any user can modify mode of file with setuid/setgid. I don't think it's correct

Reply:
the newest code still has this issue